### ISSUE TITLE
openwatcoms wstrip fails (sympton). seems to be file locking?

### ISSUE THREAD

**Issue by stu** [joncampbell123/dosbox-x#2312]
**Describe the bug**
So this is odd, compile C file, run wstrip, get cant create output file. compile C file, fail because object exists and wont overwrite...

```
LOG: Large file locking start=100000000, len=1
LOG: Large file locking start=100000000, len=1
LOG: Large file locking start=100000000, len=1
LOG: Large file locking start=100000000, len=1
LOG: Large file locking start=100000000, len=1
LOG: Warning: file /home/x/dos/c/temp/TEST.EXE exists and failed to open in write mode.
Please Remove write-protection
LOG: Large file locking start=100000000, len=1
LOG: Warning: file /home/x/dos/c/temp/TEST.EXE exists and failed to open in write mode.
Please Remove write-protection
```

and if the .OBJ file exists, when I build, it fails to overwrite with the same above error except TEST.OBJ
(but it lets me do 'del test.obj' without complaint, then I can do a build again).

**To Reproduce**
compile any C file with openwatcom v2, run wstrip on it to strip debugging information.
there may be an easier way to reproduce it...

**Environment (please complete the following information):**
 - Linux
 - DOSBox-X release version or commit SHA1 this issue occurred on 81d6c8c90c8dc56ac0e9c9bc5bf349c344eaa203

Now this was not happening when I had built off SHA1 556c084d42516e2f81df805e06ece24f2ecaed9e so must be some kind of regression maybe...

**Comment by Wengier** [comment]
Thanks for reporting this! I could reproduce the said problem on Linux. There was a bug for file opening in write mode on non-Windows systems. It is related to share functions, but not exactly the file locking code. In any case, the issue is hopefully fixed as part of #2313.

**Comment by Wengier** [comment]
This issue is already fixed in DOSBox-X 0.83.11.
